src/spark_rapids_ml/core.py

```diff
@@ -238,16 +238,7 @@ def initialize_cuml_logging(verbose: Optional[Union[bool, int]]) -> None:
             cuml_logger.set_level(log_level)
 
 
-class _CumlEstimator(Estimator, _CumlCommon, _CumlParams):
-    """
-    The common estimator to handle the fit callback (_fit). It should:
-    1. set the default parameters
-    2. validate the parameters
-    3. prepare the dataset
-    4. train and return CUML model
-    5. create the pyspark model
-    """
-
+class _CumlCaller(_CumlParams, _CumlCommon):
```

Comment:
Please add some pydoc to explain why this class exists (for future developers).

Reply:
Added some explanations.